src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OperatorIndexerCompletionProviderTests.cs

```diff
@@ -0,0 +1,500 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Completion.Providers;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Completion.CompletionProviders
+{
+    public class OperatorIndexerCompletionProviderTests : AbstractCSharpCompletionProviderTests
+    {
+        public OperatorIndexerCompletionProviderTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+        {
+
+        }
+
+        internal override Type GetCompletionProviderType()
+            => typeof(OperatorIndexerCompletionProvider);
+
+        protected override string ItemPartiallyWritten(string expectedItemOrNull) => "";
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+
+        public async Task ExplicitUserDefinedConversionIsSuggestedAfterDot()
+        {
+            await VerifyItemExistsAsync(@"
+public class C
+{
+    public static explicit operator float(C c) => 0;
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        c.$$
+    }
+}
+", "(float)");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        [WorkItem(47511, "https://github.com/dotnet/roslyn/issues/47511")]
+
+        public async Task ExplicitUserDefinedConversionIsNotSuggestedIfMemberNameIsPartiallyWritten()
+        {
+            await VerifyItemIsAbsentAsync(@"
+public class C
+{
+    public void fly() { }
+    public static explicit operator float(C c) => 0;
+}
+
+public class Program
+{
+    public void Main()
+    {
+        var c = new C();
+        c.fl$$
+    }
+}
+", "(float)");
+        }
```

Comment:
i actually don't think this is desirable behavior.  i can see someone coming to this and saying "i know there's the conversion, i want to complete to it" and have it update the code to `((float)c).`

Reply:
That complicates the completion provider because more cases need to be handled. But if it works, the odd behavior described in https://github.com/dotnet/roslyn/pull/47511#issuecomment-693413791 should also be easier to fix. I found a solution (see commit ede769d0af20b7a2fec9edaf7ae32912686f987f), but I'm not very satisfied with it (hard to understand and seems flaky to me).